### PR TITLE
Update numpy build constraints for numpy 1.25

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

Starting in numpy 1.25 (see
https://github.com/numpy/numpy/releases/tag/v1.25.0, https://numpy.org/doc/1.25/dev/depending_on_numpy.html), the numpy C API is backwards-compatible by default.

For python 3.9+, we should be able to drop the specific numpy build requirements and use `numpy>=1.25`, which is currently backwards-compatible to `numpy>=1.19`.

In the future when python 3.8 is dropped, the specific python <3.9 requirements could be dropped and the lower numpy pin could correspond to the oldest supported version for the current lower python pin.

### Types of change

?

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
The CI does not currently really test this, since numpy isn't downgraded after building.

From what I understand the lower numpy build constraint will need to be `numpy>=2.0` once numpy 2 is released.

File: spacy/tests/package/test_requirements.py [hunk 1]
```diff
@@ -22,6 +22,7 @@ def test_build_dependencies():
     ]
     # ignore language-specific packages that shouldn't be installed by all
     libs_ignore_setup = [
+        "numpy",
```
Comment (rmitsch) [inline spacy/tests/package/test_requirements.py:25]: Why is `numpy` in `libs_ignore_requirements` and `libs_ignore_setup` now?

File: setup.cfg [hunk 2]
```diff
@@ -32,8 +32,12 @@ project_urls =
 zip_safe = false
 include_package_data = true
 python_requires = >=3.6
+# NOTE: This section is superseded by pyproject.toml will be removed in spaCy v4
```
Comment (rmitsch) [inline setup.cfg:35]: ?
```suggestion
# NOTE: This section is superseded by pyproject.toml and will be removed in spaCy v4
```

File: spacy/tests/package/test_requirements.py (hunk 1 above)
Comment (adrianeboyd) [reply spacy/tests/package/test_requirements.py:25]: The test can't deal with the duplicate entries.

Comment (adrianeboyd) [reply spacy/tests/package/test_requirements.py:25]: Probably also not the diverging versions in build vs. install requirements, which we didn't have before.

**Review by rmitsch — APPROVED** [review]

**Pull request merged**